scripts/js/queries.js

```diff
@@ -40,39 +42,85 @@ function getDnssecConfig() {
   });
 }
 
+// Fetch database info (earliest timestamp, sizes, ...) from the API and
+// initialize related globals.
+function getDatabaseInfo() {
+  $.getJSON(document.body.dataset.apiurl + "/info/database", data => {
+    // earliest_timestamp is provided in seconds since epoch
+    // We have two sources: earliest_timestamp_disk (on-disk) and earliest_timestamp (in-memory)
+    // Use whichever is smallest and non-zero
+    let diskTimestamp = Number(data.earliest_timestamp_disk);
+    let memoryTimestamp = Number(data.earliest_timestamp);
+    
+    // Filter out zero/invalid timestamps
+    let validTimestamps = [diskTimestamp, memoryTimestamp].filter(ts => ts > 0);
+    
+    // Use the smallest valid timestamp, or null if none exist
+    beginningOfTime = validTimestamps.length > 0 ? Math.min(...validTimestamps) : null;
+    
+    // Round down to nearest 5-minute segment (300 seconds) if valid
+    if (beginningOfTime !== null) {
+      beginningOfTime = Math.floor(beginningOfTime / 300) * 300;
+    }
```

Comment:
Fixing linter errors:

```suggestion
    const diskTimestamp = Number(data.earliest_timestamp_disk);
    const  memoryTimestamp = Number(data.earliest_timestamp);

    // Filter out zero/invalid timestamps
    const validTimestamps = [diskTimestamp, memoryTimestamp].filter(ts => ts > 0);

    // Use the smallest valid timestamp, or null if none exist
    beginningOfTime = validTimestamps.length > 0 ? Math.min(...validTimestamps) : null;

    // Round down to nearest 5-minute segment (300 seconds) if valid
    if (beginningOfTime !== null) {
      beginningOfTime = Math.floor(beginningOfTime / 300) * 300;
    }
```